gitlab/__init__.py

```diff
@@ -0,0 +1,140 @@
+import asyncio
+
+import traceback
+
+from asyncio import Task
+
+from typing import List, Type, Awaitable
+
+from aiohttp.web import Response, Request
+
+from maubot.handlers import event, web
+
+from mautrix.types import (EventType, EventID, MessageType,
+                           TextMessageEventContent, Format)
+
+from mautrix.util.config import BaseProxyConfig, ConfigUpdateHelper
+
+from maubot import Plugin, MessageEvent
+
+from maubot.matrix import parse_markdown
+
+from .gitlab_hook import EventParse
+
+
+class Config(BaseProxyConfig):
+    def do_update(self, helper: ConfigUpdateHelper) -> None:
+        helper.copy("path")
+        helper.copy("port")
+        helper.copy("secret")
+        helper.copy("base_command")
+        helper.copy("send_as_notice")
+
+
+class Gitlab(Plugin):
+
+    def send_gitlab_event(self, room: str, msg: str) -> Awaitable[EventID]:
+        if self.config['send_as_notice']:
+            msgtype = MessageType.NOTICE
+        else:
+            msgtype = MessageType.TEXT
+
+        content = TextMessageEventContent(msgtype=msgtype,
+                                          body=msg
+                                          )
+        content.format = Format.HTML
+        content.body, content.formatted_body = parse_markdown(content.body,
+                                                              allow_html=True
+                                                              )
+        return self.client.send_message_event(room,
+                                              EventType.ROOM_MESSAGE,
+                                              content
+                                              )
+
+    async def process_hook(self, req: Request) -> None:
+        if not req.has_body:
+            self.log.debug('no body')
+            return
+
+        body = await req.json()
+
+        if 'X-Gitlab-Event' not in req.headers:
+            self.log.debug('missing X-Gitlab-Event Header')
+            return None
+
+        try:
+            GitlabEvent = EventParse[req.headers['X-Gitlab-Event']].deserialize(body)  # noqa: E501
+
+            msg = GitlabEvent.handle()
+        except Exception as e:
+            self.log.debug(traceback.format_exc())
```

Comment:
```suggestion
            self.log.debug("Failed to handle Gitlab event", exc_info=True)
```

then the self.log.debug(e) and import traceback should be removed too

Reply:
ups yes still there from debuging